matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/auth/registration/DefaultRegistrationWizard.kt

```diff
@@ -203,4 +202,18 @@ internal class DefaultRegistrationWizard(
         val session = sessionCreator.createSession(credentials, pendingSessionData.homeServerConnectionConfig)
         return RegistrationResult.Success(session)
     }
+
+    override suspend fun registrationAvailable(userName: String): RegistrationAvailability {
+        val availability = try {
+            registerAvailableTask.execute(userName)
+        } catch (exception: Throwable) {
+            if(exception.isRegistrationAvailabilityError()) {
+                return RegistrationAvailability.NotAvailable(exception as Failure.ServerError)
+            } else {
+                throw exception
```

Comment:
The treatment of the Exception could maybe be moved to the task `RegisterAvailableTask` which will return `RegistrationAvailability` instead of `Availability`

Reply:
done